src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -372,4 +373,145 @@ public void deleteInventory(MedicalInventory medicalInventory) throws OHServiceE
+		if (!chargeMovements.isEmpty()) {
+			insertedMovements.addAll(movStockInsertingManager.newMultipleChargingMovements(chargeMovements, chargeReferenceNumber));
+		}
+		if (!dischargeMovements.isEmpty()) {
+			insertedMovements.addAll(movStockInsertingManager.newMultipleDischargingMovements(dischargeMovements, dischargeReferenceNumber));
+		}
+		String status = InventoryStatus.done.toString();
+		inventory.setStatus(status);
+		this.updateMedicalInventory(inventory, false);
+		return insertedMovements;
+	}
+
+	private void checkReference(MedicalInventory medicalInventory) throws OHServiceException {
+		List<OHExceptionMessage> errors = new ArrayList<>();
+		String reference = medicalInventory.getInventoryReference();
+		String chargeReferenceNumber = reference + "-charge";
+		String dischargeReferenceNumber = reference + "-discharge";
+		boolean existWithSuffixCharge = movStockInsertingManager.refNoExists(chargeReferenceNumber);
+		boolean existWithSuffixDischarge = movStockInsertingManager.refNoExists(dischargeReferenceNumber);
+		MedicalInventory inventory = this.getInventoryByReference(reference);
+		if (existWithSuffixCharge || existWithSuffixDischarge || (inventory != null && inventory.getId() != medicalInventory.getId())) {
+			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.inventory.referencealreadyused.msg")));
+		}
+		if (!errors.isEmpty()) {
+			throw new OHServiceException(errors);
+		}
+	}
+
+	/**
+	 * Actualize the {@link MedicalInventory}.
+	 *
+	 * @param inventory - The {@link MedicalInventory}
+	 * @return {@link MedicalInventory}. It could be {@code null}.
+	 * @throws OHDataValidationException
+	 */
+	public MedicalInventory actualizeMedicalInventoryRow(MedicalInventory inventory) throws OHServiceException {
+		LocalDateTime movFrom = inventory.getInventoryDate();
+		LocalDateTime movTo = TimeTools.getNow();
+		// Fetch all the inventory row of that inventory
+		int id = inventory.getId();
+		List<MedicalInventoryRow> inventoryRowList = medicalInventoryRowManager.getMedicalInventoryRowByInventoryId(id);
+		// TODO: To decide if to make allMedicals parameter
+		boolean allMedicals = true;
+		List<Movement> movs = new ArrayList<>();
+		List<Medical> inventoryMedicalsList = inventoryRowList.stream()
+						.map(MedicalInventoryRow::getMedical)
+						.distinct()
+						.collect(Collectors.toList());
+		if (allMedicals) {
+			// Fetch all movements without filtering by medical code
+			movs.addAll(movBrowserManager.getMovements(null, null, null, null, movFrom, movTo, null, null, null, null));
+		} else {
+			// Fetch only movements concerning inventoryRowSearchList list
+			for (Medical medical : inventoryMedicalsList) {
+				movs.addAll(movBrowserManager.getMovements(medical.getCode(), null, null, null, movFrom, movTo, null, null, null, null));
+			}
+		}
+		// Get all the lot of the movements
+		List<Lot> lotOfMovements = movs.stream().map(m -> m.getLot()).collect(Collectors.toList());
```

Comment:
```suggestion
		List<Lot> lotOfMovements = movs.stream().map(Movement::getLot).collect(Collectors.toList());
```
Replace lambda with method reference